pkg/daemon/ceph/client/pool.go

```diff
@@ -529,7 +529,11 @@ func extractPoolDetails(rule ruleSpec) (string, string) {
 			failureDomain = step.Type
 		}
 		if step.ItemName != "" {
-			deviceClass = step.ItemName
+			// we are not using crushRoot currently, but in future improve
+			// this part to improve advanced crushrule support
+			crushRootAndDeviceClass := step.ItemName
+			parts := strings.Split(crushRootAndDeviceClass, "~")
+			deviceClass = parts[1]
```

Comment:
Are there guaranteed to have two parts separated by the tilde? I didn't think so, and perhaps this is causing the CI to fail.

Reply:
thanks for the taking a look Travis, will fix it